linkerd/app/gateway/src/gateway.rs

```diff
@@ -54,11 +59,30 @@ where
         match self {
             Self::Outbound {
                 ref mut outbound,
+                ref local_identity,
                 ref forwarded_header,
             } => {
+                // Check forwarded headers to see if this request has already
+                // transited through this gateway.
+                for fwd in request
+                    .headers()
+                    .get_all(http::header::FORWARDED)
+                    .into_iter()
+                    .filter_map(|h| h.to_str().ok())
+                {
+                    if let Some(by) = fwd_by(fwd) {
+                        if by == local_identity.as_ref() {
+                            tracing::info!(fwd, "Loop detected");
```

Comment:
should this be a warning? also, do we want to consider eventually adding a metric label or something so that the responses that fail due to looping are counted separately? of course, this can be a follow up; just a thought.

Reply:
Alternatively, we could throw an error here and let the error-handler take the wheel...

Reply:
https://github.com/linkerd/linkerd2-proxy/compare/ver/error-status?expand=1